waku/waku_rln_relay/group_manager/on_chain/group_manager.nim

```diff
@@ -477,10 +475,25 @@ method init*(g: OnchainGroupManager): Future[void] {.async.} =
                        "could not get the membership deposit: " & getCurrentExceptionMsg())
   g.membershipFee = some(membershipFee)
 
+  var deployedBlockNumber: Uint256
+  try:
+    deployedBlockNumber = await rlnContract.deployedBlockNumber().call()
+  except CatchableError:
+    raise newException(ValueError, 
+                       "could not get the deployed block number: " & getCurrentExceptionMsg())
+  g.rlnContractDeployedBlockNumber = some(cast[BlockNumber](deployedBlockNumber))
+
   ethRpc.ondisconnect = proc() =
     error "Ethereum client disconnected"
-    let fromBlock = g.latestProcessedBlock.get()
+    let fromBlock = max(g.latestProcessedBlock.get(), g.rlnContractDeployedBlockNumber.get())
     info "reconnecting with the Ethereum client, and restarting group sync", fromBlock = fromBlock
+    try:
+      let newEthRpc = waitFor newWeb3(g.ethClientUrl)
```

Comment:
maybe we should have a maximum number of reconnection attempts here to prevent connecting to a dead node forever - can address in a follow up pr

Reply:
yep, mind tracking it?

Not that concern about infinite reconections as long as the node is aware that "its not ready" (eg via the `/health` endpoint) so not a high prio.

But perhaps we need some delay? to avoid an infinite connect-disconnect-connect loop?

Reply:
yup max-retries + backoff